macros/src/types/named.rs

```diff
@@ -75,16 +76,29 @@ fn format_field(
         return Ok(());
     }
 
-    let (ty, optional_annotation) = match optional {
-        true => (extract_option_argument(&field.ty)?, "?"),
-        false => (&field.ty, ""),
+    if let (Some(_type_as), Some(_type_override)) = (&type_as, &type_override) {
+        syn_err!("`type` is not compatible with `as`")
+    }
+
+    let parsed_ty = if let Some(_type_as) = &type_as {
```

Comment:
Rename the variable to remove the `_` prefix, and if possible use `ref` instead of `&`